app/lib/collections/OrgLogs.js

```diff
@@ -1 +1,85 @@
-App.Collections.OrgLogs = new Mongo.Collection('orgLogs');
\ No newline at end of file
+var OrgLogsSchema = new SimpleSchema({
+  timestamp: {
+    type: Date,
+    optional: false
+  },
+  url: {
+    type: String,
+    optional: false
+  },
+  statusCode: {
+    type: Number,
+    optional: true
+  },
+  error: {
+    type: Object,
+    optional: true,
+    blackbox: true
+  },
+  lastSuccess: {
+    type: Date,
+    optional: true
+  },
+  data: {
+    type: Object,
+    optional: true,
+    blackbox: true
+  },
+  successCount: {
+    type: Number,
+    optional: true,
+    defaultValue: 0
+  },
+  failureCount: {
+    type: Number,
+    optional: true,
+    defaultValue: 0
+  }
+});
+
+OrgLogs = new Mongo.Collection('orgLogs');
+OrgLogs.attachSchema(OrgLogsSchema);
+
+function createNewSuccessLog(url, data) {
+  return { 
+    statusCode: 200, 
+    url: url, 
+    data: data, 
+    error: null, 
+    timestamp: new Date(),
+    lastSuccess: new Date()
+  };
+}
+
+function createNewErrorLog(url, e) {
+  var statusCode = null;
+  if (e.response) {
+      statusCode = e.response.statusCode;
+  } 
+  return { 
+    timestamp: new Date(),
+    statusCode: statusCode, 
+    url: url, 
+    data: null, 
+    // clone to plain object to get rid of tricky native properties
+    error: JSON.parse(JSON.stringify(e))
```

Comment:
> Beware using the JSON.parse(JSON.stringify(obj)) method on Date objects - JSON.stringify(new Date()) returns a string representation of the date in ISO format, which JSON.parse() doesn't convert back to a Date object.

[http://stackoverflow.com/questions/122102/what-is-the-most-efficient-way-to-deep-clone-an-object-in-javascript/5344074#5344074](source)